publications/tasks.py

```diff
@@ -1,27 +1,73 @@
-    schedule_type=Schedule.DAILY
-)
 
-def new_manuscript_updates():
-    """
-    send email for manuscript updates
-    """
-    one_minute_ago = timezone.now() - timezone.timedelta(minutes=1)
-    new_manuscripts = Publication.objects.filter(
-        date = one_minute_ago
-    )
-    subject = 'New Manuscripts'
-    email = "test@example.com"
-    message =f"""\ Hello, we send you updated manuscripts that you were looking for:    {new_manuscripts}   """
-    send_mail(subject, message, from_email= "optimetageo@gmail.com",recipient_list=[email])
+def harvest_data(url):
+    # url1 = os.path.join(os.getcwd(),'harvesting\\journal_1\\oai_dc.xml') # forlocalfile
+    response = requests.get(url)
+    DOMTree = xml.dom.minidom.parseString(response.content)# parse xml as DOM
+    #DOMTree = xml.dom.minidom.parse(url1) 
+    collection = DOMTree.documentElement
+    articles = collection.getElementsByTagName("dc:identifier")                                            
+    no_articles = len(articles)  # number of articles in journal
+    for i in range(no_articles):
+        identifier = collection.getElementsByTagName("dc:identifier")
+        identifier_value = identifier[i].firstChild.nodeValue
+        if identifier_value.startswith('https://'):
+            link_value = identifier_value               
+            #get geometry from html
+            #identifier_url = os.path.join(os.getcwd(),'harvesting\\journal_1\\article_01.html')
+            geom = get_geom(link_value)
+            geom_data = geom["features"][0]["geometry"]
+            geom_data_string = json.dumps(geom_data)
+            # preparing geometry data in accordance to geosAPI fields
+            type_geom= {'type': 'GeometryCollection'}
+            geom_content = {"geometries" : [geom_data_string]}
+            type_geom.update(geom_content)
+            geom_data = json.dumps(type_geom)
+            # geometry field accept json object as string
+            geom_object = GEOSGeometry(geom_data) #GeometryCollection object
+            #bounds=MultiPolygon([Polygon(((-117.869537353516, 33.5993881225586),(-117.869537353516, 33.7736549377441),(-117.678024291992, 33.7736549377441),(-117.678024291992, 33.5993881225586),(-117.869537353516, 33.5993881225586)))])           
+            #gc = GeometryCollection(Point(0, 0), MultiPoint(Point(0, 0), Point(1, 1)), bounds)
+            title = collection.getElementsByTagName("dc:title")
+            title_value = title[0].firstChild.nodeValue
+            abstarct = collection.getElementsByTagName("dc:description") 
+            abstarct_text = abstarct[0].firstChild.nodeValue
+            journal = collection.getElementsByTagName("dc:publisher")
+            journal_value = journal[0].firstChild.nodeValue
+            date = collection.getElementsByTagName("dc:date")
+            date_value = date[0].firstChild.nodeValue 
+            #get latest id from table
+            h = Publication.objects.all()
```

Comment:
Please do not do this ID incrementing manually, you can just let Django manage the primary field in the database, see https://docs.djangoproject.com/en/4.1/topics/db/models/#automatic-primary-key-fields

Alternatively, you could use a UUID as the unique ID, but I don't think that's needed for our use.

Reply:
ID field is automatically added by Django. Retracted my incrementation.